File: RedSheet.DbEntities/Models/vTheirRequirementDetailRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTheirRequirementDetailRecords", Schema = "dbo")]
	[Serializable]
	public partial class vTheirRequirementDetailRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string ldo { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string mdo { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectRequirementId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Requirement { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SortOrder { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int TheirRequirementDetailId { get; set; } 

	public vTheirRequirementDetailRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-their-requirement-detail-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTheirRequirementDetailRecord {
    constructor(vTheirRequirementDetailRecord?: vTheirRequirementDetailRecord )  {
        let properties = [ "ldo", "mdo", "projectRequirementId", "requirement", "sortOrder", "theirRequirementDetailId",];
        for (let property of properties)
            if (vTheirRequirementDetailRecord && vTheirRequirementDetailRecord[property])
                this[property] = vTheirRequirementDetailRecord[property];
    }
 
    @required()
    @maxLength(1000)
	ldo : string =   undefined;
 
    @required()
    @maxLength(1000)
	mdo : string =   undefined;
 
	projectRequirementId : number =   0 ;
 
    @required()
    @maxLength(1000)
	requirement : string =   undefined;
 
    @range(1,2147483647)
	sortOrder : number =   undefined;
 
    @range(1,2147483647)
	theirRequirementDetailId : number =   undefined;


}
